more_itertools/more.py

```diff
@@ -2437,11 +2437,10 @@ class islice_extended:
     """
 
     def __init__(self, iterable, *args):
-        it = iter(iterable)
         if args:
-            self._iterable = _islice_helper(it, slice(*args))
+            self._iterable = _islice_helper(iterable, slice(*args))
```

Comment:
Why do we need to change variable names here (and below)? I'd rather not have this in the diff.

Reply:
Previously, an iterator was extracted from `iterable` to the `it` variable, and then this `it` was passed to `_islice_helper`. Now, `iterable` itself is passed to `_islice_helper`. This is important because if we pass an iterator, we won't be able to determine that it belongs to a `Sized` iterable, and we will not be able to apply optimization. For this reason, I changed the name of the argument from `it` to `iterable` so that it could be seen that an iterable is passed, not an iterator. But some conditional branches need an iterator of iterable. Therefore, in these branches I get an iterator and put it in the variable `iterator'. If we had left `it', it might have been confusing. Should I rename `iterator` to `it`?